plugin-server/src/main/ingestion-queues/session-recording/session-recordings-consumer.ts

```diff
@@ -339,47 +332,30 @@ export class SessionRecordingIngester {
                     func: async () => {
-                        const parsedMessages: IncomingRecordingMessage[] = []
-                        for (const message of messages) {
-                            const { partition, offset, timestamp } = message
-
-                            this.partitionMetrics[partition] = this.partitionMetrics[partition] || {}
-                            const metrics = this.partitionMetrics[partition]
-
-                            // If we don't have a last known commit then set it to the offset before as that must be the last commit
-                            metrics.lastMessageOffset = offset
-                            // For some reason timestamp can be null. If it isn't, update our ingestion metrics
-                            metrics.lastMessageTimestamp = timestamp || metrics.lastMessageTimestamp
-
-                            counterKafkaMessageReceived.inc({ partition })
-
-                            const recordingMessage = await parseKafkaMessage(
-                                message,
-                                (token) =>
-                                    this.teamsRefresher.get().then((teams) => ({
-                                        teamId: teams[token]?.teamId || null,
-                                        consoleLogIngestionEnabled: teams[token]?.consoleLogIngestionEnabled ?? true,
-                                    })),
-                                this.sharedClusterProducerWrapper
-                            )
-
-                            if (recordingMessage) {
-                                parsedMessages.push(recordingMessage)
+                        const { sessions, partitionStats } = await parseKafkaBatch(
+                            messages,
+                            (token) =>
+                                this.teamsRefresher.get().then((teams) => ({
+                                    teamId: teams[token]?.teamId || null,
+                                    consoleLogIngestionEnabled: teams[token]?.consoleLogIngestionEnabled ?? true,
+                                })),
+                            this.sharedClusterProducerWrapper
+                        )
+                        recordingMessages = sessions
+                        for (const partitionStat of partitionStats) {
+                            const metrics = this.partitionMetrics[partitionStat.partition] ?? {}
+                            metrics.lastMessageOffset = partitionStat.offset
+                            if (partitionStat.timestamp) {
+                                // Could be empty on Kafka versions before KIP-32
+                                metrics.lastMessageTimestamp = partitionStat.timestamp
                             }
+                            this.partitionMetrics[partitionStat.partition] = metrics
                         }
-
-                        recordingMessages = reduceRecordingMessages(parsedMessages)
```

Comment:
Ah-ha, I'm new to this code, it took me a second to find the `cloneObject` was in here.